lib/irida/pipeline.rb

```diff
@@ -84,28 +91,30 @@ def process_property(key, name, property, required)
     end
 
     def process_samplesheet_schema
-      JSON.parse(schema_input_loc.read)
+      default_schema = JSON.parse(schema_input_loc.read)
+      default_schema.deep_merge!(@samplesheet_schema_overrides_for_entry)
+      default_schema
     end
 
     def show_section?(properties)
       properties.values.any? { |property| !property.key?('hidden') }
     end
 
-    def overrides_for_entry(entry)
-      return {} if entry['versions'].nil?
+    def overrides_for_entry(entry, version)
+      return {} if version.nil?
+
+      overrides = entry['overrides'].deep_dup || {}
 
-      overrides = entry['overrides'] || {}
-      version_overrides = entry['versions'].find do |version|
-        version['name'] == @version
-      end || {}
       overrides
-        .deep_merge(
-          version_overrides.key?('overrides') ? version_overrides['overrides'] : {}
+        .deep_merge!(
+          version.key?('overrides') ? version['overrides'] : {}
         )
+
+      overrides
     end
 
-    def default_workflow_params_for_entry(entry) # rubocop:disable Metrics/CyclomaticComplexity
-      return {} if entry['versions'].nil?
+    def default_workflow_params_for_entry # rubocop:disable Metrics/CyclomaticComplexity
+      return {} if @overrides.nil?
```

Comment:
Suggestion (blocking): You can get rid of this return statement as from above `@overrides` will always at least be an empty hash.

Reply:
Removed in e165362d7e29ae127e011b58492019004f9b0c57